### ISSUE TITLE
Observable.retry and Observable.retryWith memory leak in Zone.js

### ISSUE THREAD

**Issue by Guardiannw** [ReactiveX/RxJS#1400]
@trxcllnt I have found another memory leak in Angular's Zone.js (I assume) when using Observable.retry and Observable.retryWith.  This memory leak is there even after implementing your PR #1395.  See [my repo](https://github.com/Guardiannw/RxJS).  I have tested the issue and confirmed that it does not leak whenever it is used in isolation, so it must be going through some loops like Observable.interval that Zone.js is tracking.  Do you know of the internals off the top of your head as to reasons I might be having the leak with these operators?

Reference: 
#1391 
#1394 
#1395 
Thanks,

**Comment by blesh** [comment]
So for the same reasons I'm closing #1391 I'm closing this Issue. See comment here for more detail: https://github.com/ReactiveX/RxJS/issues/1391#issuecomment-190451469

TL;DR: This was an issue with zone.js that is fixed in zone.js and awaiting integration into Angular 2.

**Comment by blesh** [comment]
cc/ @mhevery

**Comment by mhevery** [comment]
The reason why Zone.js has a memory leak is that it forks on each async task, which eventually runs out of memory. As of this morning Zone.js has been rewritten, and the memory leak removed, so it should no longer be an issue.

**Comment by Guardiannw** [comment]
@mhevery Great! Do you have any idea when it will be integrated into ng2?
